version: '3.8'

services:
  api:
    build:
      context: .
      dockerfile: Dockerfile.test
    environment:
      - NODE_ENV=test
      - RUNNING_IN_DOCKER=true
      - TEST_DB_HOST=postgres-test
      - TEST_DB_PORT=5433
      - TEST_DB_USERNAME=postgres
      - TEST_DB_PASSWORD=postgres
      - TEST_DB_NAME=arbi_test
    ports:
      - "8081:8081"
    depends_on:
      postgres-test:
        condition: service_healthy
    command: [ "npm", "run", "test:api" ]
    volumes:
      - ./:/app
      - node_modules:/app/node_modules

  postgres-test:
    image: postgres:16
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: arbi_test
    ports:
      - "5433:5433"
    command: -p 5433
    volumes:
      - postgres-test-data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres -p 5433" ]
      interval: 5s
      timeout: 5s
      retries: 5

volumes:
  node_modules:
  postgres-test-data:
